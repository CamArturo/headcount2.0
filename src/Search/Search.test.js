import './Search.css';
import React from 'react';
import Search from './Search';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Search Tests', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Search />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toHaveLength(1);
  });
  it('Should call HandleSearchEvent onChange', () => {
    const props = {
      handleSearchEvent: jest.fn()
    };
    const mockEvent = {target: {value: 'col'}};
    const component = shallow(<Search {...props} />);

    component.find('input').simulate('change', mockEvent);
    expect(props.handleSearchEvent).toHaveBeenCalled();
    expect(props.handleSearchEvent).toBeCalledWith(mockEvent);
  });
});