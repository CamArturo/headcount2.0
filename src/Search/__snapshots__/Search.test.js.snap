// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Tests matches the snapshot 1`] = `
<div
  className="card-container"
>
  <form
    action=""
    className="district-form"
  >
    <h1>
      Welcome to HeadCount 2.0
    </h1>
    <input
      className="district-input"
      onChange={[Function]}
      placeholder="Search Districts"
      type="text"
    />
  </form>
  <hr />
</div>
`;
